ros_ws/src/autonomous/src/wall.cpp

```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
 
 float Wall::getAngle()
 {
-    return atan((m_range1 * cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
-                sin(std::abs(m_angle1 - m_angle2)));
+    return atan((m_range1 * (float)cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
+                (float)sin(std::abs(m_angle1 - m_angle2)));
 }
 
 float Wall::predictDistance(float distancce_to_current_position)
 {
     float angle = this->getAngle();
-    float currentWallDistance = m_range2 * cos(angle);
-    return currentWallDistance + distancce_to_current_position * sin(angle);
+    float currentWallDistance = m_range2 * (float)cos(angle);
+    return currentWallDistance + distancce_to_current_position * (float)sin(angle);
```

Comment:
```suggestion
    return currentWallDistance + distancce_to_current_position * std::sin(angle);
```
Because `sin(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen.